edtLHD/src/main/java/fr/univtln/lhd/model/entities/dao/slots/SubjectDAO.java

```diff
@@ -25,7 +26,7 @@ public class SubjectDAO implements DAO<Subject> {
      * Constructor of SubjectDao, initiate connection and prepared statement
      * @throws SQLException throw a SQLException if there is a problem with the connection or database prepared statement
      */
-    private SubjectDAO() throws SQLException {
+    public SubjectDAO() throws SQLException {
```

Comment:
This  basically negates the whole idea of a connection pool, as the connection you will be using is gonna be on hold even when no operation is pending (its lifetime follows that of the class)

What you should instead do is 
1. make each of those static and or final strings 
2.  Make each of the method throw an Exception (whatever really)
3. in each method, use a try with ressources to get the conn. and the prepare statement :  the following works : 
```java 
//this.<whatever> := "Select from Subject where id = (?) "
//...
try( 
      Connection connection = Datasource.getConnection(); // I've added this in the Datasource in an early PR
       PreparedStatement stmt = connection.prepareStatement(this.<whatever>) // no ; required on the last one
       ) 
       {  /*resultSet.Set(...) or Get*/ }
catch (SqlException e) { throw e; /* or whatever */}

I see this is applicable to other DAOs as well, I can provide help if you so wish.